torchrl/collectors/collectors.py

```diff
@@ -759,18 +771,18 @@ def _step_and_maybe_reset(self) -> None:
                 _reset = None
                 td_reset = None
             td_reset = self.env.reset(td_reset)
-            self._tensordict.update(td_reset, inplace=True)
-            done = self._tensordict.get("done")
-            if (_reset is None and done.any()) or (
-                _reset is not None and done[_reset].any()
-            ):
-                raise RuntimeError(
-                    f"Env {self.env} was done after reset on specified '_reset' dimensions. This is (currently) not allowed."
-                )
             traj_done_or_terminated = done_or_terminated.sum(
                 tuple(range(self._tensordict.batch_dims, done_or_terminated.ndim)),
                 dtype=torch.bool,
             )
+            self._tensordict.get_sub_tensordict(traj_done_or_terminated).update(
+                td_reset[traj_done_or_terminated], inplace=True
+            )
+            done = self._tensordict[traj_done_or_terminated].get("done")
+            if (_reset is None and done.any()) or (_reset is not None and done.any()):
```

Comment:
This check here might be worth keeping like before because
`self._tensordict.get("done")[_reset]`
checks allt the done_sec dims, while
`self._tensordict[traj_done_or_terminated].get("done")`
uses only `traj_done_or_terminated` which is smaller than the _reset

Reply:
taking your example if `_reset.shape = [3, 4, 5]` but only `_reset[0,0,0] = True`, when you compute `traj_done_or_terminated` you get `[True, False,False]`.

now when you do `self._tensordict[traj_done_or_terminated].get("done")` you get a done of shape `[4,5]` and you check that none of it is True, while you should check only `[0,0]`

Reply:
I agree with that point but at that point we'll need a deep refactoring of that method anyway and this line is likely to be changed no?
I'm not against the idea of going back to the previous version but I'd like to understand why one is better than the other

Reply:
Yep they are equivalent, and also equvalent to `not self._tensordict.get("done").any()` which may be the simplest

Reply:
I was thinking the same thing
What about: we take the simplest option (for now) and leave it as is until we figure out what to do with envs that can start with a done or env that keep being executed when done?

Reply:
The if in this pr has an or which can be removed BTW dunno if you saw it

Reply:
sorry I did not get that